src/app/header/header.component.ts

```diff
@@ -9,17 +10,29 @@ import { Context } from './../models/context';
 import { ContextService } from '../shared/context.service';
 import { DummyService } from './../shared/dummy.service';
 
+
+interface MenuHiddenCallback {
+  (context: Context, menuItem: MenuItem): boolean;
+}
+
 @Component({
   selector: 'alm-app-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss'],
   providers: []
 })
-
 export class HeaderComponent implements OnInit {
   title = 'Almighty';
   imgLoaded: Boolean = false;
 
+  menuCallbacks = new Map<String, MenuHiddenCallback>([
```

Comment:
For flow sake, I think functions should be below the constructor. It's not a big deal.